include/mbedtls/config.h

```diff
@@ -2206,6 +2206,23 @@
  */
 #define MBEDTLS_AESNI_C
 
+/**
+ * \def MBEDTLS_CUSTOM_HAS_AESNI
+ *
+ * Link to an externally defined `mbedtls_aesni_has_support` function. By
+ * default mbedtls uses the `cpuid` instruction to determine whether AES-NI
+ * is supported at run time.
+ *
+ * Module:  library/aesni.c
+ * Caller:  library/gcm.c
+ *          library/aes.c
+ *
+ * Requires: MBEDTLS_HAVE_X86_64
+ *
+ * Uncomment to provide a custom `mbedtls_aesni_has_support` function.
+ */
+//#define MBEDTLS_CUSTOM_HAS_AESNI
```

Comment:
The option should be named `MBEDLTS_AESNI_HAS_SUPPORT_ALT`, on the same model as `MBEDTLS_AES_ENCRYPT_ALT` etc. It's not only a convention for users, but also options that require third-party code need special handling in our test scripts unless they end in `_ALT`.